testplan/common/utils/testing.py

```diff
@@ -413,3 +415,30 @@ def copy(self):
         typed_values attribute.
         """
         return FixMessage(self.items(), typed_values=self.typed_values)
+
+
+def calculate_pool_size_by_tasks(
+    tasks: Collection[Task], plan_runtime_target: int
+):
+    """
+    Determine the number of pool size needed for running all the targets
```

Comment:
Calculate the right size of the pool based on the weight (runtime) of the tasks, so that runtime of all tasks meets the plan_runtime_target.